tests/unit/test_fine_grained_filters.py

```
@@ -0,0 +1,315 @@
+    assert marker.ops is None
+    assert marker.apply is None
+
+
+def test_filter_field_with_ops_normalizes_to_tuple() -> None:
+    marker = Strawchemy("sqlite").filter_field(ops=["eq", "in", "like"])
+    assert marker.ops == ("eq", "in", "like")
+    assert marker.apply is None
+
+
+def test_filter_field_with_apply_captures_callable_and_join() -> None:
+    def _apply(statement: Any, *_args: Any, **_kwargs: Any) -> Any:
```

Ruff (0.15.17)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `statement`

(ANN401)